doc/bn.tex

```diff
@@ -820,28 +820,32 @@ \subsection{Single Digit}
 \}
 \end{alltt} \end{small}
 
-\subsection{Long Constants}
+\subsection{Int32 and Int64 Constants}
 
-To set a constant that is the size of an ISO C ``unsigned long'' and larger than a single digit the following function
-can be used.
+To set a constant with 32 or 64 bits the following function can be used.
 
-\index{mp\_set\_int}
+\index{mp\_set\_sint}
 \begin{alltt}
-int mp_set_int (mp_int * a, unsigned long b);
+void mp_set_sint (mp_int * a, int32_t b);
+void mp_set_uint (mp_int * a, uint32_t b);
+void mp_set_sint64 (mp_int * a, int64_t b);
+void mp_set_uint64 (mp_int * a, uint64_t b);
 \end{alltt}
 
-This will assign the value of the 32-bit variable $b$ to the mp\_int $a$.  Unlike mp\_set() this function will always
-accept a 32-bit input regardless of the size of a single digit.  However, since the value may span several digits
-this function can fail if it runs out of heap memory.
-
-To get the ``unsigned long'' copy of an mp\_int the following function can be used.
+This will assign the value of the variable $b$ to the mp\_int $a$. The unsigned variants set a positive value, the signed variants set the sign. The value can be obtained again by calling the following function.
```

Comment:
I would keep it shorter, just: "This will assign the sign and value of the input \texttt{b} to \texttt{mp\_int a}."

"the following functions" (the plural "s" has been cut of in your text again)

Reply:
done